### PR TITLE
[MI-2356] Updated docs

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
1. Added info to add "User-Agent" header.
2. Added info to set value of "va.bot.to.bot.take.control_times".

### CODE REVIEW

File: docs/developer_docs.md [hunk 1]
```diff
@@ -87,6 +87,11 @@ This plugin integrates the ServiceNow Virtual Agent in Mattermost. It is created
     - [Set up trusted media domains](https://docs.servicenow.com/bundle/quebec-now-intelligence/page/administer/virtual-agent/task/ccif-secure-file-upload.html)
       - [Configuring the form layout](https://docs.servicenow.com/en-US/bundle/sandiego-platform-administration/page/administer/form-administration/concept/configure-form-layout.html)
 
+- **Invalid response**
+    We get an "undefined" response from Virtual Agent API in a specific case. You can read [this](https://www.servicenow.com/community/virtual-agent-nlu-forum/getting-improper-response-from-virtual-agent-bot-integration-api/m-p/255032) for more information.
```
Comment (Kshitij-Katiyar) [inline docs/developer_docs.md:91]: May be this is better?
```suggestion
 If we get an "undefined" response from Virtual Agent API in some specific cases. Please refer to [this](https://www.servicenow.com/community/virtual-agent-nlu-forum/getting-improper-response-from-virtual-agent-bot-integration-api/m-p/255032) for more information.
```

File: docs/servicenow_setup.md [hunk 2]
```diff
@@ -19,7 +19,25 @@ We need to set up the virtual agent on a ServiceNow instance, to which we will s
...
+  - Adding "User-Agent" header:
+    Add the "User-Agent" header as shown in the screenshot below with the value "ServiceNow"
+    ![image](https://user-images.githubusercontent.com/55234496/201832569-9f11f919-b7c9-4192-a9cf-89a955da08c1.png)
+
+
+    **Note-** If the "insert a new row..." (as shown in the screenshot below) option is not visible to you in the "HTTP Request" section:
+
+    ![image](https://user-images.githubusercontent.com/55234496/201840807-f593a0cf-aa7a-4f34-bf29-4956f8b680e3.png)
+
+      1. Change the application scope to "Virtual Agent API". (From top right corner).
```
Comment (Kshitij-Katiyar) [inline docs/servicenow_setup.md:31]:
```suggestion
      1. Change the application scope to "Virtual Agent API". (From the top right corner).
```

File: docs/servicenow_setup.md [hunk 3]
```diff
@@ -19,7 +19,25 @@ We need to set up the virtual agent on a ServiceNow instance, to which we will s
...
+
+          ![image](https://user-images.githubusercontent.com/55234496/201833135-7907cdbc-5e00-4338-b81d-c48204eae614.png)
+
+      2. Make sure that "Enable list edit" and "Double click to edit" option are checked in the "HTTP header setting".
```
Comment (ayusht2810) [inline docs/servicenow_setup.md:35]:
```suggestion
      2. Make sure that the "Enable list edit" and "Double click to edit" options are checked in the "HTTP header setting".
```

**Review by ayusht2810 — APPROVED** [review]

File: docs/developer_docs.md (hunk 1 above)
Comment (raghavaggarwal2308) [reply docs/developer_docs.md:91]: Changed the second line. But I think the first line if correct as we are telling a developer about the issue.

**Review by Kshitij-Katiyar — APPROVED** [review]

File: docs/developer_docs.md [hunk 4]
```diff
@@ -87,6 +87,11 @@ This plugin integrates the ServiceNow Virtual Agent in Mattermost. It is created
     - [Set up trusted media domains](https://docs.servicenow.com/bundle/quebec-now-intelligence/page/administer/virtual-agent/task/ccif-secure-file-upload.html)
       - [Configuring the form layout](https://docs.servicenow.com/en-US/bundle/sandiego-platform-administration/page/administer/form-administration/concept/configure-form-layout.html)
 
+- **Invalid response**
+    We get an "undefined" response from Virtual Agent API in a specific case. Please refer to [this](https://www.servicenow.com/community/virtual-agent-nlu-forum/getting-improper-response-from-virtual-agent-bot-integration-api/m-p/255032) for more information.
+
+    [va.bot.to.bot.take.control_times](https://docs.servicenow.com/bundle/sandiego-servicenow-platform/page/administer/virtual-agent/concept/bot2bot.html)
```
Comment (manojmalik20) [inline docs/developer_docs.md:93]:
```suggestion
    We get an "undefined" response from Virtual Agent API in a specific case. Please refer to [this](https://www.servicenow.com/community/virtual-agent-nlu-forum/getting-improper-response-from-virtual-agent-bot-integration-api/m-p/255032) community question for more information about the issue. To bypass this issue, you should increase the value of `va.bot.to.bot.take.control_times` variable. You can read more about it [here](https://docs.servicenow.com/bundle/sandiego-servicenow-platform/page/administer/virtual-agent/concept/bot2bot.html).
```

File: docs/servicenow_setup.md [hunk 5]
```diff
@@ -19,7 +19,25 @@ We need to set up the virtual agent on a ServiceNow instance, to which we will s
     https://<your-mattermost-url>/plugins/mattermost-plugin-servicenow-virtual-agent/api/v1/nowbot/processResponse?secret=<your-webhook-secret>
     ```
     **Note**: (Webhook secret can be generated from the Mattermost system console settings of the Virtual agent plugin.)
- 
+  - Adding "User-Agent" header:
```
Comment (manojmalik20) [inline docs/servicenow_setup.md:22]:
```suggestion
  - Adding "User-Agent" header:
  
```

Comment (manojmalik20) [reply docs/servicenow_setup.md:22]: You can see this link and add the reason for adding this - 
https://support.servicenow.com/kb?id=kb_article_view&sysparm_article=KB0720934
https://www.servicenow.com/community/it-service-management-forum/403-forbidden-username-password-combo/m-p/770873/page/2

File: docs/servicenow_setup.md [hunk 6]
```diff
@@ -19,7 +19,8 @@ We need to set up the virtual agent on a ServiceNow instance, to which we will s
     https://<your-mattermost-url>/plugins/mattermost-plugin-servicenow-virtual-agent/api/v1/nowbot/processResponse?secret=<your-webhook-secret>
     ```
     **Note**: (Webhook secret can be generated from the Mattermost system console settings of the Virtual agent plugin.)
-  - Adding "User-Agent" header:
+  - Adding "User-Agent" header:([Reason](https://support.servicenow.com/kb?id=kb_article_view&sysparm_article=KB0720934))
```
Comment (manojmalik20) [inline docs/servicenow_setup.md:22]:
```suggestion
  - Adding "User-Agent" header: ([Reason](https://support.servicenow.com/kb?id=kb_article_view&sysparm_article=KB0720934))
```

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**